app/models/partner_external_account.rb

```diff
@@ -0,0 +1,39 @@
+class PartnerExternalAccount < ApplicationRecord
+  belongs_to :partner
+
+  encrypts :info, type: :json
+  encrypts :sub
+  blind_index :sub
+
+  validates :sub, presence: true, uniqueness: {scope: :provider_id, message: " déjà lié à un espace professionnel Covidliste"}
+  validates :provider_id, presence: true, inclusion: {in: PartnerExternalAccountProvider::Providers::ALL.keys.map(&:to_s)}
+  validates :info, presence: true
+
+  def service_name
+    if info.present?
+      info["service_name"].to_s
+    end
```

Comment:
:question: Why the `to_s`?

:question: Is there any case where `service_name` wouldn't be present?

:thinking: Maybe this could make sense :man_shrugging: 

```suggestion
    info.try(:[], "service_name")
```